test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs

```diff
@@ -550,7 +550,7 @@ void ConfigureTestServices(IServiceCollection services)
             Assert.Equal(shouldEnrichBeCalled, enrichCalled);
         }
 
-        [Fact(Skip = "Changes pending on instrumentation")]
+        [Fact]
```

Comment:
also add a test where user manually overrides Activity.Current. In that case, we wont find the asp.net core activity.

Reply:
not sure if I follow - Is there a way to manually override the `Activity.Current` without making existing `Activity.Current`  as parent.

Reply:
One way you achieve this is in the custom middleware like

```csharp
public override void PreProcess(HttpContext context)
{
    Activity.Current = null;
    this.activity = this.activitySource.StartActivity(this.activityName);
}
``

Reply:
I think the key principle we need to follow here is:
Optimize for normal path.
Users can mess with things (like set Activity.Current = null etc.), and we can chose to *not* collect traces in those situations.
We should try not to modify the activity created by user. (Checking operation name is a good option. Users can still mess with it, by create Activity with operationname, and even with same activitysource as the asp.net core framework.)

Reply:
Makes sense.